packages/langchain/src/orchestration/client.ts

```diff
@@ -108,14 +112,86 @@ export class OrchestrationClient extends BaseChatModel<
 
     const content = res.getContent();
 
-    // TODO: Add streaming as soon as we support it
     await runManager?.handleLLMNewToken(
       typeof content === 'string' ? content : ''
     );
 
     return mapOutputToChatResult(res.data);
   }
 
+  /**
+   * Stream response chunks from the OrchestrationClient.
+   * @param messages - The messages to send to the model.
+   * @param options - The call options.
+   * @param runManager - The callback manager for the run.
+   * @returns An async generator of chat generation chunks.
+   */
+  override async *_streamResponseChunks(
+    messages: BaseMessage[],
+    options: typeof this.ParsedCallOptions,
+    runManager?: CallbackManagerForLLMRun
+  ): AsyncGenerator<ChatGenerationChunk> {
+    const controller = new AbortController();
+    if (options.signal) {
+      options.signal.addEventListener('abort', () => controller.abort());
+    }
+
+    const orchestrationMessages =
+      mapLangchainMessagesToOrchestrationMessages(messages);
+
+    const { inputParams, customRequestConfig } = options;
+    const mergedOrchestrationConfig = this.mergeOrchestrationConfig(options);
+
+    const orchestrationClient = new OrchestrationClientBase(
+      mergedOrchestrationConfig,
+      this.deploymentConfig,
+      this.destination
+    );
+
+    const response = await this.caller.call(() =>
+      orchestrationClient.stream(
+        // Todo Adapt messagesHistory after: https://github.com/SAP/ai-sdk-js-backlog/issues/293
+        { messagesHistory: orchestrationMessages, inputParams },
+        controller,
+        options.streamOptions,
+        customRequestConfig
+      )
+    );
+
+    for await (const chunk of response.stream) {
+      const messageChunk = mapOrchestrationChunkToLangChainMessageChunk(chunk);
```

Comment:
[req] If we want to implement this correctly with the multi choices support, here we should yield `generationChunk` for each choice. So wrap the below content with

```
chunk.data.orchestration_result?.choices?.forEach(choice => {
    ...
}); // Or export getChoices() and use that
```

And then when calling functions like `getDeltaContent()`, give it the choice index.

The current implementation only support n=1 and will keep yielding chunks with empty content if choice index is not 0.